### PR TITLE
Support Batching in the custom Formatter

### PR DESCRIPTION
**Pull request by abhilashmandaliya**
* Expose an interface for batching incoming XDR records before sending them to the destination
* Created ElasticsearchOutboundMetadata for Elasticseatch outbound connector
* Set the sourceCompatibility to 1.8
* Fixed the Javadoc error
* Integrated Snyk Vulnerability scanning

### CODE REVIEW

**Review by abhilashmandaliya — COMMENTED** [review]
LGTM :+1:

File: src/main/java/com/aerospike/connect/outbound/format/BatchItem.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+@AllArgsConstructor
+@EqualsAndHashCode
+@Getter
+@ToString
+public class BatchItem<T extends OutboundMetadata> {
```
Comment (tejassp) [inline src/main/java/com/aerospike/connect/outbound/format/BatchItem.java:32]: @abhilashmandaliya please add Javadoc for this class.

File: src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     *     BytesMessage.
+     *     </li>
+     *     <li>{@link TextOutboundRecord} for JMS outbound destination when
+     *     the JMS message should be sent as a JMS TextMessage.</li>
+     *     <li>{@link SkipOutboundRecord} to skip dispatching the change
+     *     notification record to the outbound destination.</li>
+     * </ul>
+     *
+     * <p>
+     * When an exception is thrown by this method, the records are acknowledged
```
Comment (tejassp) [inline src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java:63]: Should this be changed from "records are acknowledged" to "all the records in the batch are acknowledged"?

File: src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * destination, etc.
+ *
+ * @param <T> the metadata associated with the outbound records.
+ */
+public interface BatchFormatter<T extends OutboundMetadata> {
+    /**
+     * Format records into a custom format.
+     *
+     * <p>
+     * This interface is identical to the {@link Formatter} interface. Here we
```
Comment (tejassp) [inline src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java:44]: Do not use "we", "us", etc. Here it should probably read "collection of BatchItems are passed as a parameter to the formatter" or something along those lines.

File: src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+package com.aerospike.connect.outbound.format;
+
+import com.aerospike.connect.outbound.esp.EspOutboundMetadata;
+import com.aerospike.connect.outbound.pulsar.PulsarOutboundMetadata;
+import lombok.NonNull;
+
+import java.util.List;
+
+/**
+ * Format the Aerospike records into an outbound records for the outbound
```
Comment (tejassp) [inline src/main/java/com/aerospike/connect/outbound/format/BatchFormatter.java:28]: Remove the an from "an outbound records", it is plural not singular.

**Review by tejassp — COMMENTED** [review]
Javadoc changes

**Pull request merged**